code/modules/antagonists/wraith/critters/exploder.dm

```diff
@@ -69,7 +69,8 @@
 					continue
 				else
 					boutput(M, SPAN_ALERT("You are sprayed with disgusting rotting flesh! You're pretty sure some of it got in your mouth."))
-			M.emote("scream")
+			if (!is_dead_or_ghost_role(M))
```

Comment:
```suggestion
			if (isalive(M) && !isingtangible(M))
```
This makes more sense, ghosts are dead but not intangible and blob overminds/AI eyes etc are alive but intangible. `is_dead_or_ghost_role` includes ghostdrones, critter respawns etc and is mostly used for respawning code, not determining if something is real and exists nearby.

Reply:
This makes more sense! I appreciate the explanation, it helps a ton. Thank you so much!